messaging/src/main/java/org/axonframework/eventhandling/deadletter/jpa/JpaSequencedDeadLetterQueue.java

```diff
@@ -0,0 +1,716 @@
+            entityManager().persist(deadLetter);
+        });
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void evict(DeadLetter<? extends M> letter) {
+        if (!(letter instanceof JpaDeadLetter)) {
+            throw new WrongDeadLetterTypeException(
+                    "Evict should be called with a JpaDeadLetter instance. Instead got: [" + letter.getClass().getName()
+                            + "]");
+        }
+        JpaDeadLetter<M> jpaDeadLetter = (JpaDeadLetter<M>) letter;
+
+        transactionManager.executeInTransaction(
+                () -> entityManager().createQuery("delete from DeadLetterEntry dl where dl.deadLetterId=:deadLetterId")
+                                     .setParameter("deadLetterId", jpaDeadLetter.getId())
+                                     .executeUpdate());
+    }
+
+    private EntityManager entityManager() {
+        return entityManagerProvider.getEntityManager();
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void requeue(@Nonnull DeadLetter<? extends M> letter,
+                        @Nonnull Function<DeadLetter<? extends M>, DeadLetter<? extends M>> letterUpdater)
+            throws NoSuchDeadLetterException {
+        if (!(letter instanceof JpaDeadLetter)) {
+            throw new WrongDeadLetterTypeException(
+                    "Evict should be called with a JpaDeadLetter instance. Instead got: [" + letter.getClass().getName()
+                            + "]");
+        }
+        EntityManager entityManager = entityManager();
+        DeadLetter<? extends M> updatedLetter = letterUpdater.apply(letter).markTouched();
+        String id = ((JpaDeadLetter<? extends M>) letter).getId();
+        DeadLetterEntry letterEntity = entityManager.find(DeadLetterEntry.class,
+                                                          id);
+        if (letterEntity == null) {
+            throw new NoSuchDeadLetterException("Can not find dead letter with id [" + id + "]");
+        }
+        letterEntity.setDiagnostics(updatedLetter.diagnostics(), serializer);
+        letterEntity.setLastTouched(updatedLetter.lastTouched());
+        letterEntity.setCause(updatedLetter.cause().orElse(null));
+        letterEntity.clearProcessingStarted();
+        entityManager.persist(letterEntity);
+    }
+
+    @Override
+    public boolean contains(@Nonnull Object sequenceIdentifier) {
+        String identifier = toIdentifier(sequenceIdentifier);
+        return getNumberInQueue(identifier) > 0;
+    }
+
+    @Override
+    public Iterable<DeadLetter<? extends M>> deadLetterSequence(@Nonnull Object sequenceIdentifier) {
+        String identifier = toIdentifier(sequenceIdentifier);
+
+        return entityManagerProvider
+                .getEntityManager()
+                .createQuery(
+                        "SELECT dl FROM DeadLetterEntry dl where dl.processingGroup=:processingGroup and dl.sequence=:identifier",
```

Comment:
Nit: I'd expect the SQL to follow one format. Either none of the keywords are capitalized or all of them are capitalized.